### PR TITLE
5 create namespace and organizationalunit models

### LINKED ISSUE #5: Create Namespace and OrganizationalUnit models
OrganizationalUnit
* Name

Namespace
* one_to_one OrganizationalUnit
* one_to_many Repositories

### PR DESCRIPTION
**Pull request by eugenk**
This shall fix #5. It changes an old migration, but I think as long as we don't have a productive system, this is okay.

### CODE REVIEW

File: app/models/namespace.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+# The namespace is used to group repositories by the owner and to allow setting
+# repository-local permissions for teams and users.
+class Namespace < Sequel::Model
+  plugin :timestamps
+
+  many_to_one :organizational_unit
```
Comment (phyrog) [inline app/models/namespace.rb]: Shouldn't this be `one_to_one`? I think we should separate the namespace that belongs to a user/team and the ones a user/team has access to. If we separate it, a user/team can only ever have one namespace that belongs to it.

File: db/migrate/20160919140517_create_namespaces.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    create_table :namespaces do
+      primary_key :id
+      foreign_key :organizational_unit_id, :organizational_units, index: :uniqe
```
Comment (phyrog) [inline db/migrate/20160919140517_create_namespaces.rb]: I think this should be `:unique` instead.

File: app/models/organizational_unit.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# Superclass of Team and User
+class OrganizationalUnit < Sequel::Model
```
Comment (phyrog) [inline app/models/organizational_unit.rb]: We could include the `class_table_inheritance` plugin and the `type`/`kind` column already I think.

Comment (phyrog) [inline app/models/organizational_unit.rb]: Do we want to use the `association_dependencies` plugin to automatically delete the namespace when a OrgUnit is destroyed?

**Comment by eugenk** [comment]
This should be done now (by commit amend).

**Pull request merged**